device/iot-device-samples/pnp-device-sample/thermostat-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/Thermostat.java

```diff
@@ -98,6 +168,60 @@ public void run() {
         }).start();
     }
 
+    private static void initializeAndProvisionDevice() throws ProvisioningDeviceClientException, IOException, URISyntaxException, InterruptedException {
+        SecurityProviderSymmetricKey securityClientSymmetricKey = new SecurityProviderSymmetricKey(deviceSymmetricKey.getBytes(), registrationId);
+        ProvisioningDeviceClient provisioningDeviceClient = null;
+        ProvisioningStatus provisioningStatus = new ProvisioningStatus();
+
+        provisioningDeviceClient = ProvisioningDeviceClient.create(globalEndpoint, scopeId, provisioningProtocol, securityClientSymmetricKey);
+
+        AdditionalData additionalData = new AdditionalData();
+        additionalData.setProvisioningPayload(String.format("{\"modelId\": \"%s\"}", MODEL_ID));
+
+        provisioningDeviceClient.registerDevice(new ProvisioningDeviceClientRegistrationCallbackImpl(), provisioningStatus, additionalData);
+
+        while (provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() != ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_ASSIGNED)
+        {
+            if (provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_ERROR ||
+                    provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_DISABLED ||
+                    provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_FAILED)
+            {
+                provisioningStatus.exception.printStackTrace();
+                System.out.println("Registration error, bailing out");
+                break;
+            }
+            System.out.println("Waiting for Provisioning Service to register");
+            Thread.sleep(MAX_TIME_TO_WAIT_FOR_REGISTRATION);
```

Comment:
No need to sleep 10 seconds each time. 1 second should be fine. Also rename this to POLLING_PERIOD or something similar